src/plugins/data_views/common/data_views/data_view_lazy.ts

```diff
@@ -0,0 +1,619 @@
+      scriptFields[field.name] = {
+        script: {
+          source: field.script as string,
+          lang: field.lang,
+        },
+      };
+    });
+
+    return {
+      scriptFields,
+      docvalueFields,
+      runtimeFields: this.getRuntimeMappings(),
+    };
+  }
+
+  getRuntimeMappings(): estypes.MappingRuntimeFields {
+    // @ts-expect-error composite type is not yet supported by es client but it can be forced
+    return this.runtimeFieldMap;
+  }
+
+  /**
+   * Creates static representation of the data view.
+   * @param includeFields Whether or not to include the `fields` list as part of this spec. If not included, the list
+   * will be fetched from Elasticsearch when instantiating a new Data View with this spec.
+   */
+  public async toSpec(params?: { fieldParams?: GetFieldsParams }): Promise<DataViewSpec> {
+    const spec = this.toSpecShared(!!params?.fieldParams);
+
+    if (params?.fieldParams) {
+      const fields: DataViewFieldSpecMap = {};
+      const fieldMap = (await this.getFields(params.fieldParams)).getFieldMap();
+      Object.values(fieldMap).forEach((field) => {
+        fields[field.name] = field.toSpec({
+          getFormatterForField: this.getFormatterForField.bind(this),
+        });
+      });
+      spec.fields = fields;
+    }
+
+    return spec;
+  }
+
+  /**
+   * Creates a minimal static representation of the data view. Fields and popularity scores will be omitted.
+   */
+  public toMinimalSpec(): Omit<DataViewSpec, 'fields'> {
+    // removes `fields`
+    const spec = this.toSpecShared(false);
+
+    if (spec.fieldAttrs) {
+      // removes `count` props (popularity scores) from `fieldAttrs`
+      spec.fieldAttrs = pickBy(
+        mapValues(spec.fieldAttrs, (fieldAttrs) => omit(fieldAttrs, 'count')),
+        (trimmedFieldAttrs) => Object.keys(trimmedFieldAttrs).length > 0
+      );
+
+      if (Object.keys(spec.fieldAttrs).length === 0) {
+        delete spec.fieldAttrs;
+      }
+    }
+
+    return Object.fromEntries(Object.entries(spec).filter(([, v]) => typeof v !== 'undefined'));
+  }
+
+  /**
+   * returns true if dataview contains TSDB fields
+   */
+  async isTSDBMode() {
+    const fieldMap = (await this.getFields()).getFieldMap();
+
+    return Object.values(fieldMap).some(
+      (field) => field.timeSeriesDimension || field.timeSeriesMetric
+    );
+  }
+
+  removeScriptedField(fieldName: string) {
+    this.deleteScriptedFieldInternal(fieldName);
+    this.fieldCache.delete(fieldName);
+  }
+
+  upsertScriptedField(field: FieldSpec) {
+    return this.upsertScriptedFieldInternal(field);
+  }
+
+  isTimeBased = async () => !!(await this.getTimeField());
+
+  async isTimeNanosBased(): Promise<boolean> {
+    const field = await this.getTimeField();
+    return field ? field.type === 'date_nanos' : false;
+  }
+
+  getTimeField = () => (this.timeFieldName ? this.getFieldByName(this.timeFieldName) : undefined);
```

Comment:
```suggestion
  getTimeField = async () => (this.timeFieldName ? this.getFieldByName(this.timeFieldName) : undefined);
```
It might be worth making this method async to return `Promise<DataViewField | undefined>` instead of `Promise<DataViewField> | undefined` which seems trickier for consumers.

Reply:
They're equivalent constructs. Awaiting non promises simply returns the item being awaited.

Reply:
I was thinking more if a consumer is using promises since they'd have to conditionally call `.then()`, but it's a minor nit and also not breaking if we decided to change it later, so I think it's fine to leave as is.

Reply:
Ah, good point